### PR TITLE
Add scan manager main services with database interaction

### PR DESCRIPTION
**Pull request by sacheeramesh**
## Purpose
There is a requirement to automate the security scans and govern the security process for the WSO2 products. scan manager will manage the all scan request.

## Goals
managing the security scans.

## Approach
expose the APIs to the front-end for getting the new scan request then manage them and initiate scans in actual scanners. expose the API to scanners to communicate with front-end through the scan-manager 

## User stories
API exposed to front-end can archive the following operations

1.start scan
2.stop scan
3.get scan report

APIs exposed to the scanner  can archive the following operations

1.update the Scan status
2.log 

## Release note
added the database interactions, modified the models and project structure, implements the basic operations of scan manager

## Documentation
design specification - https://docs.google.com/document/d/1RWzMr8ax9hrQxGdC7kUhOhW_-T8LjzjShtxMjLho99Q/edit?usp=sharing

## Automation tests
 - Unit tests 
   done
 - Integration tests
   done

## Security checks

   -Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
   -Ran FindSecurityBugs plugin and verified report? yes
   -Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/config/ScanMangerConfigurationBuilder.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.security.tools.scanmanager.config;
+
+
```
Comment (NShani) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/config/ScanMangerConfigurationBuilder.java]: remove additional new line.

File: internal/scan-manager/pom.xml [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-databind</artifactId>
+      <version>2.9.7</version>
+    </dependency>
+    <dependency>
+      <groupId>javax.servlet</groupId>
+      <artifactId>jstl</artifactId>
+      <version>1.2</version>
+    </dependency>
+    <!--<dependency>-->
```
Comment (NShani) [inline internal/scan-manager/pom.xml]: Remove unused dependencies.

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/config/ScanMangerConfigurationBuilder.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+ * Scan manager configuration builder class.
+ */
+public class ScanMangerConfigurationBuilder {
+
+    private static final String SCAN_MANAGER_CONFIG_FILE = "scan-manager-config.yaml";
+
+    /**
+     * Reading the scan manager configuration from scanner-config.yaml.
+     *
+     * @return,
```
Comment (NShani) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/config/ScanMangerConfigurationBuilder.java]: add return object's description.

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/config/ScanMangerConfigurationBuilder.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            throw new ScanManagerException("Unable to find the scan manager configuration");
+        }
+        return scanManagerConfiguration;
+    }
+
+    /**
+     * Reading the scan manager configuration from a given config file.
+     *
+     * @param configFile
+     * @return
```
Comment (NShani) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/config/ScanMangerConfigurationBuilder.java]: add return object's description.

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/config/ScanManagerConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (NShani) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/config/ScanManagerConfiguration.java]: change license header to 2019.

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/dao/LogDAOImpl.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
+package org.wso2.security.tools.scanmanager.dao;
+
+
```
Comment (NShani) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/dao/LogDAOImpl.java]: Remove additional new line.

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/dao/ScanDAO.java [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package org.wso2.security.tools.scanmanager.dao;
+
+import org.wso2.security.tools.scanmanager.model.Scan;
+/**
+ * The class {@code } is the DAO class that persist data on the SCAN table.
+ */
+public interface ScanDAO {
+    boolean persist(Scan scan);
```
Comment (NShani) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/dao/ScanDAO.java]: Add new line between two methods.

File: internal/scan-manager/src/main/resources/scan-manager-config.yaml [hunk 8]
```diff
@@ -0,0 +1,26 @@
+#Scan Manager Configuration
+
+databaseUrl : jdbc:mysql://localhost:3306/smdb?useSSL=false
+userName : root
```
Comment (NShani) [inline internal/scan-manager/src/main/resources/scan-manager-config.yaml]: remove these values and keep placeholders.

File: internal/scan-manager/src/main/resources/scan-manager-config.yaml [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+password : admin
```
Comment (NShani) [inline internal/scan-manager/src/main/resources/scan-manager-config.yaml]: remove this value and keep placeholders.

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/dao/LogDAOImpl.java [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.stereotype.Repository;
+import org.wso2.security.tools.scanmanager.controller.ScanController;
+import org.wso2.security.tools.scanmanager.model.Log;
+/**
+ * The class {@code LogDAOImpl} is the DAO class that manage the persistence method implementations of the,
+ * Scan logs.
+ */
+@Repository
+public class LogDAOImpl implements LogDAO {
+    private static final Logger logger = LoggerFactory.getLogger(ScanController.class);
```
Comment (kasundharmadasa) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/dao/LogDAOImpl.java]: Add the correct class name

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/ScanRequest.java [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.security.tools.scanmanager.model;
+
+import java.io.Serializable;
+import java.util.Map;
+/**
```
Comment (kasundharmadasa) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/ScanRequest.java]: Add a new line after imports

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/Scan.java [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+    private String reportPath;
+
+    @Column(name = "STATUS")
+    private String status;
+
+    @Column(name = "SUBMITTED_TIMESTAMP")
+    private java.sql.Timestamp submittedTimestamp;
+
+    @Column(name = "ACTUAL_SCANNER_ID")
+    private String actualScannerIdl;
```
Comment (kasundharmadasa) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/Scan.java]: Do we need the last character ("l") for this variable name

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/ScanStatus.java [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.security.tools.scanmanager.model;
+/**
+ * Model Enum to represent a Status of the Scans.
+ */
+public enum ScanStatus {
+    PRNDING, SUBMITTED, RUNNING, CANCELED, COMPLETED, FINISHED, ERROR, FAILED
```
Comment (kasundharmadasa) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/ScanStatus.java]: Enum should be changed to PENDING

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/config/ScanMangerConfigurationBuilder.java (hunk 1 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/config/ScanMangerConfigurationBuilder.java]: done

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/config/ScanMangerConfigurationBuilder.java (hunk 3 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/config/ScanMangerConfigurationBuilder.java]: done

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/config/ScanMangerConfigurationBuilder.java (hunk 4 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/config/ScanMangerConfigurationBuilder.java]: done

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/config/ScanManagerConfiguration.java (hunk 5 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/config/ScanManagerConfiguration.java]: done

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/dao/LogDAOImpl.java (hunk 6 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/dao/LogDAOImpl.java]: done

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/dao/ScanDAO.java (hunk 7 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/dao/ScanDAO.java]: done

File: internal/scan-manager/src/main/resources/scan-manager-config.yaml (hunk 8 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/resources/scan-manager-config.yaml]: done

File: internal/scan-manager/src/main/resources/scan-manager-config.yaml (hunk 9 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/resources/scan-manager-config.yaml]: done

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/dao/LogDAOImpl.java (hunk 10 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/dao/LogDAOImpl.java]: done

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/ScanRequest.java (hunk 11 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/ScanRequest.java]: done

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/Scan.java (hunk 12 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/Scan.java]: no, I removed it

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/ScanStatus.java (hunk 13 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/ScanStatus.java]: done

**Comment by sacheeramesh** [comment]
suggested modifications

File: internal/scan-manager/pom.xml (hunk 2 above)
Comment (sacheeramesh) [reply internal/scan-manager/pom.xml]: done

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/dao/LogDAOImpl.java (hunk 10 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/dao/LogDAOImpl.java]: done

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/controller/CallbackController.java [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+
+/**
+ * The class {@code ScannerEngineController} is the web controller which defines the routines for managing
+ * scans and combine scan containers to scan manager.
+ */
+@Controller
+@RequestMapping("callback")
+public class CallbackController {
+
+    private static final Logger logger = LoggerFactory.getLogger(ScanController.class);
```
Comment (kasundharmadasa) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/controller/CallbackController.java]: add the proper class for the logger

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/exception/ScanManagerException.java [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.security.tools.scanmanager.exception;
+/**
```
Comment (kasundharmadasa) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/exception/ScanManagerException.java]: add a new line before the class comment

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/exception/ScanManagerException.java [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+ * The class {@code ScanManagerException} extends {@link Exception}, is the general Scan Manager exception,
+ * for all operations dealing with Scan Manager.
+ */
+public class ScanManagerException extends Exception {
+    /**
```
Comment (kasundharmadasa) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/exception/ScanManagerException.java]: add a new line before the method comment

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/Field.java [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.security.tools.scanmanager.model;
+/**
```
Comment (kasundharmadasa) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/Field.java]: add a new line before the class comment

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/Scan.java [hunk 18]
```diff
@@ -0,0 +1,187 @@
...
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+/**
+ * Model class to represent a scan.
+ */
+
```
Comment (kasundharmadasa) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/Scan.java]: additional newline after class comment.

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/Scan.java [hunk 19]
```diff
@@ -0,0 +1,187 @@
...
+    @Column(name = "STATUS")
+    private String status;
+
+    @Column(name = "SUBMITTED_TIMESTAMP")
+    private java.sql.Timestamp submittedTimestamp;
+
+    @Column(name = "ACTUAL_SCANNER_ID")
+    private String actualScannerId;
+
+
```
Comment (kasundharmadasa) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/Scan.java]: unnecessary new .ine

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/service/LogServiceImpl.java [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.wso2.security.tools.scanmanager.dao.LogDAO;
+import org.wso2.security.tools.scanmanager.model.Log;
+
+/**
+ * The class {@code LogService} is the service class that manage the method implementations of the
+ * Scan logs.
+ */
+
```
Comment (kasundharmadasa) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/service/LogServiceImpl.java]: unnecessary new line.

File: internal/scan-manager/src/main/webapp/WEB-INF/web.xml [hunk 21]
```diff
@@ -0,0 +1,23 @@
...
+      <param-name>contextConfigLocation</param-name>
+      <param-value>/WEB-INF/dispatcher-servlet.xml</param-value>
+    </init-param>
+    <load-on-startup>1</load-on-startup>
+  </servlet>
+  <servlet-mapping>
+    <servlet-name>dispatcher</servlet-name>
+    <url-pattern>/</url-pattern>
+  </servlet-mapping>
+
```
Comment (kasundharmadasa) [inline internal/scan-manager/src/main/webapp/WEB-INF/web.xml]: Unnecessary new line.

File: internal/scan-manager/src/main/webapp/WEB-INF/web.xml [hunk 22]
```diff
@@ -0,0 +1,23 @@
...
```
Comment (kasundharmadasa) [inline internal/scan-manager/src/main/webapp/WEB-INF/web.xml]: Unnecessary new line.

File: internal/scan-manager/src/main/webapp/WEB-INF/dispatcher-servlet.xml [hunk 23]
```diff
@@ -0,0 +1,75 @@
...
+    </bean>
+
+    <bean id="myTransactionManager"
+          class="org.springframework.orm.hibernate5.HibernateTransactionManager">
+        <property name="sessionFactory" ref="sessionFactory" />
+    </bean>
+
+    <tx:annotation-driven transaction-manager="myTransactionManager" />
+
+</beans>
```
Comment (kasundharmadasa) [inline internal/scan-manager/src/main/webapp/WEB-INF/dispatcher-servlet.xml]: Please add a new line

File: internal/scan-manager/src/main/resources/scan-manager-config.yaml [hunk 24]
```diff
@@ -0,0 +1,26 @@
...
+ftpPort :
+ftpUsername :
+ftpPassword :
+queueUsername :
+queuePassword :
+brokerHostname : 
+brokerPort : 
+
+
+
```
Comment (kasundharmadasa) [inline internal/scan-manager/src/main/resources/scan-manager-config.yaml]: Unnecessary new line.

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/exception/ScanManagerExceptionHandler.java [hunk 25]
```diff
@@ -0,0 +1,43 @@
...
+
+    @ResponseBody
+    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
+    @ExceptionHandler
+    ErrorMessage exceptionHandler(ScanManagerException e) {
+        return new ErrorMessage(HttpStatus.INTERNAL_SERVER_ERROR.value(), e.getMessage());
+    }
+}
+
+
```
Comment (kasundharmadasa) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/exception/ScanManagerExceptionHandler.java]: Unnecessary new lines.

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/exception/ScanManagerExceptionHandler.java [hunk 26]
```diff
@@ -0,0 +1,43 @@
...
+/**
+ * Global Exception handler class for Scan Manager exceptions.
+ */
+@ControllerAdvice
+public class ScanManagerExceptionHandler {
+
+    @ResponseBody
+    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
+    @ExceptionHandler
+    ErrorMessage exceptionHandler(ScanManagerException e) {
```
Comment (kasundharmadasa) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/exception/ScanManagerExceptionHandler.java]: Please add a method description.

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/exception/ScanManagerException.java [hunk 27]
```diff
@@ -0,0 +1,52 @@
...
+     * Constructs a new runtime exception with the specified detail message and
+     * cause.
+     */
+    public ScanManagerException(String message, Throwable e) {
+        super(message, e);
+    }
+}
+
+
+
```
Comment (kasundharmadasa) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/exception/ScanManagerException.java]: Unnecessary new lines.

File: internal/scan-manager/pom.xml [hunk 28]
```diff
@@ -0,0 +1,213 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
```
Comment (kasundharmadasa) [inline internal/scan-manager/pom.xml]: Unnecessary new line

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/dao/LogDAOImpl.java [hunk 29]
```diff
@@ -0,0 +1,36 @@
...
+ * Scan logs.
+ */
+@Repository
+public class LogDAOImpl implements LogDAO {
+    private static final Logger logger = LoggerFactory.getLogger(LogDAOImpl.class);
+
+    @Autowired
+    private SessionFactory sessionFactory;
+
+    @Override
```
Comment (kasundharmadasa) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/dao/LogDAOImpl.java]: Better to add a method description.

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/controller/CallbackController.java (hunk 14 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/controller/CallbackController.java]: done

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/exception/ScanManagerException.java (hunk 15 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/exception/ScanManagerException.java]: done

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/exception/ScanManagerException.java (hunk 16 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/exception/ScanManagerException.java]: done

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/Field.java (hunk 17 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/Field.java]: done

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/Scan.java (hunk 18 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/Scan.java]: done

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/Scan.java (hunk 19 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/Scan.java]: done

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/service/LogServiceImpl.java (hunk 20 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/service/LogServiceImpl.java]: done

File: internal/scan-manager/src/main/webapp/WEB-INF/web.xml (hunk 21 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/webapp/WEB-INF/web.xml]: done

File: internal/scan-manager/src/main/webapp/WEB-INF/web.xml (hunk 22 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/webapp/WEB-INF/web.xml]: done

File: internal/scan-manager/src/main/webapp/WEB-INF/dispatcher-servlet.xml (hunk 23 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/webapp/WEB-INF/dispatcher-servlet.xml]: done

File: internal/scan-manager/src/main/resources/scan-manager-config.yaml (hunk 24 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/resources/scan-manager-config.yaml]: done

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/exception/ScanManagerExceptionHandler.java (hunk 25 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/exception/ScanManagerExceptionHandler.java]: done

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/exception/ScanManagerExceptionHandler.java (hunk 26 above)
Comment (sacheeramesh) [reply internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/exception/ScanManagerExceptionHandler.java]: done

File: internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/Log.java [hunk 30]
```diff
@@ -0,0 +1,102 @@
...
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+/**
+ * Model class to represent a Logs in the scan containers.
+ */
+
```
Comment (kasundharmadasa) [inline internal/scan-manager/src/main/java/org/wso2/security/tools/scanmanager/model/Log.java]: Redundent new line.

**Comment by ayomawdb** [comment]
Thank you @sacheeramesh for getting the comments resolved! Merging the PR.

**Pull request merged**